src/pages/formations/apprentissage/index.page.test.tsx

```diff
@@ -69,21 +96,101 @@ describe('Page Formations en Apprentissage', () => {
 		});
 
 		it('n‘a pas de défaut d‘accessibilité', async () => {
-			const analyticsService = aManualAnalyticsService();
-			mockUseRouter({});
+			mockUseRouter(
+				{
+					query: rechercheFormationAlternanceQuery(),
+				},
+			);
+			const resultats = aRésultatRechercheFormationList();
 
 			const { container } = render(
 				<DependenciesProvider
-					analyticsService={analyticsService}
-					formationService={aFormationService()}
+					analyticsService={aManualAnalyticsService()}
 					metierLbaService={aMetierService()}
 					localisationService={aLocalisationService()}
 				>
-					<FormationAlternancePage />
+					<FormationAlternancePage resultats={resultats}/>
 				</DependenciesProvider>,
 			);
 
 			await expect(container).toBeAccessible();
 		});
+
+		it('lorsque la page est affichée sans query params, retourne des props vide', async () => {
+			const result = await getServerSideProps(aGetServerSidePropsContext());
+			expect(result).toEqual({ props: {} });
+		});
+
+		describe('lorsque la page est affichée avec des query params', () => {
+			it.each([{ queryInvalide: '75056' },
+				rechercheFormationAlternanceQuery({ niveauEtudes: 'pas un niveau correcte' }),
+				rechercheFormationAlternanceQuery({ codeRomes: undefined }),
+				rechercheFormationAlternanceQuery({ distanceCommune: undefined }),
+			])('et qu‘ils sont invalides, retourne une erreur et ne fait pas de recherche', async () => {
+				const context = aGetServerSidePropsContext({ query: { queryInvalide: '75056' } });
+
+				const result = await getServerSideProps(context);
+
+				expect(result).toEqual({ props: { erreurRecherche: ErreurMetier.DEMANDE_INCORRECTE } });
+				expect(dependencies.formationDependencies.rechercherFormation.handle).not.toHaveBeenCalled();
+			});
```

Comment:
Si tu veux matcher le format de juste en dessous, tu peux aussi faire un `describe.each` normalement. Ou un `describe > it.each`